build/build/src/enso.rs

```diff
@@ -160,6 +172,8 @@ impl BuiltEnso {
                 only.iter().any(|test| test.contains("Microsoft_Tests")),
         };
 
+        let cloud_credentials_file = cloud_tests::prepare_credentials_file().await?;
```

Comment:
It seems that this temp file does not live long enough to be seen by running tests.

e.g. I can see that the path is set in env variables, but while the tests are running the file does not exist.

I assume it is successfully created, so it is probably dropped too soon.

@Frizi do you have any hints maybe what I can do to extend its lifetime until the `self.run_test` below finishes?

Reply:
Yes it is dropped at the end of the function, once the run future is returned, i.e. before the execution begins. You have to manually drop it only after the program has finished running (i.e. inside the future, past the await point).

```rs
async move {
    let result = command?.run_ok().await;
    drop(cloud_credentials_file);
    result
}

``

Reply:
Hmm, but I'm running several commands in there (it's inside `into_iter().map`), so I'm not sure if that will work?

Reply:
Right, I missed that. In this case you have to `drop` it after the `future::join_all` call, which is the next await point after the loop generating invididual futures.